spec/requests/groups_controller_spec.rb

```diff
@@ -858,6 +858,54 @@ def expect_type_to_return_right_groups(type, expected_group_ids)
 
       context "user default notifications" do
         it "should update default notification preference for existing users" do
+          user1 = Fabricate(:user)
+          user2 = Fabricate(:user)
+          group.add(user1)
+          group.add(user2)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          group_users = group.group_users
+          expect(response.parsed_body["user_count"]).to eq(group_users.count)
+
+          group_users.where(user: user1).update_all(notification_level: 1)
+
+          put "/groups/#{group.id}.json", params: {
+            group: {
+              default_notification_level: 2
+            }
+          }
+
+          expect(response.status).to eq(200)
+          expect(response.parsed_body["user_count"]).to eq(group.group_users.count - 1)
```

Comment:
Since `user1`'s group_user notification level has been updated to `1`, should we be asserting that his notification level remains at `1`?

Reply:
if `user1` notification level is updated to `2` wrongly then we will get incorrect out in `response.parsed_body["user_count"]`. Anyway, it would be better to make sure the user1's notification level after the update 👍 I will do the change.